wordpress-mobile/WordPress-Android
# [Compile Warnings As Errors] All Modules - Enable All Warnings as Errors

Parent: #17173
Closes: #17181
Closes: #17182

This PR enables all warnings as errors for all modules.

This `allWarningsAsErrors` configuration was previously applied on the module level in order to make sure that, as the overall `Kotlin Warnings as Errors` work is progressing, no new warnings are added to a specific module, which is already free of warnings.

However, now that the overall `Kotlin Warnings as Errors` work is complete on all modules, this module level `allWarningsAsErrors` configuration is now replaced by a root level `allWarningsAsErrors` configuration that will be applied by default to all modules.

-----

PS: @ovitrif I added you as the main reviewer, that is, in addition to @wordpress-mobile/apps-infrastructure team itself, but randomly, since I want someone from the WordPress Android team to primarily sign-off on that change. 🥇

-----

To test:

- There is nothing much to test here.
- Verifying that all the CI checks are successful should be enough.
- If you really want to be thorough, try to add an artificial warning per source (`main`, `test`, `androidTest`), per module (`WordPress`, `editor`, `image-editor`, `processors`, `annotations`) and see the `assemble` related task fail. Then, remove this warning and see it succeeding again. 

-----

### Merge instructions

- [x] Wait for #17287 to be approved and merged to `trunk`.

-----

## Regression Notes
1. Potential unintended areas of impact

N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)

See `To test` section above.

3. What automated tests I added (or what prevented me from doing so)

N/A

-----

PR submission checklist:
